File: ticket-service/src/main/java/com/tickets/ticket_service/ticket/application/GetMyTicketsUseCase.java
package com.tickets.ticket_service.ticket.application;

import com.tickets.ticket_service.shared.PageResult;
import com.tickets.ticket_service.shared.UseCase;
import com.tickets.ticket_service.ticket.domain.Ticket;
import com.tickets.ticket_service.ticket.domain.TicketRepository;

import java.util.UUID;

@UseCase
public class GetMyTicketsUseCase {

    private final TicketRepository ticketRepository;

    public GetMyTicketsUseCase(TicketRepository ticketRepository) {
        this.ticketRepository = ticketRepository;
    }

    public PageResult<Ticket> execute(UUID userId, int page, int size) {
        return ticketRepository.findAllByUserIdWithOrder(userId, page, size);
    }
}

File: frontend/eventFlow/src/features/tickets/application/use-cases/GetMyTicketsUseCase.ts
import type { TicketRepository } from '../../domain/ports/TicketRepository'
import type { Ticket } from '../../domain/entities/Ticket'
import type { PaginatedResponse } from '@/shared/types/pagination.types'

export class GetMyTicketsUseCase {
  constructor(private readonly ticketRepository: TicketRepository) {}

  execute(page: number, size: number): Promise<PaginatedResponse<Ticket>> {
    return this.ticketRepository.getMyTickets(page, size)
  }
}
